### PR TITLE
Distinguish network for bitmex price feed

### PR DESCRIPTION
**Pull request by da-kami**
For the `testnet` and `signet` setup we should use query the `testnet` price and not `mainnet`.

### CODE REVIEW

**Review by bonomat — APPROVED** [review]
I think this makes sense.

File: bitmex-stream/src/lib.rs [hunk 1]
```diff
@@ -63,3 +65,18 @@ pub fn subscribe<const N: usize>(
...
+pub enum Network {
+    Mainnet,
+    Testnet,
+}
+
+impl Network {
+    pub fn to_url(&self) -> String {
+        match self {
+            Network::Mainnet => "www.bitmex.com".to_string(),
+            Network::Testnet => "testnet.bitmex.com".to_string(),
```
Comment (klochowicz) [inline bitmex-stream/src/lib.rs:79]: nit: `www.` prefix only in one type of network? it seems like it's probably not needed in mainnet either?

**Review by klochowicz — APPROVED** [review]

File: bitmex-stream/src/lib.rs (hunk 1 above)
Comment (da-kami) [reply bitmex-stream/src/lib.rs:79]: I tried without and it is needed. I don't have an easy answer why.

Comment (bonomat) [reply bitmex-stream/src/lib.rs:79]: The subdomain of `www` is  not voluntarily, it's the official url they are running their API on :D 

See https://testnet.bitmex.com/api/explorer/ vs https://www.bitmex.com/api/explorer/

**Pull request merged**